workshops/urls.py

```diff
@@ -176,22 +176,22 @@
     url(r'^training_requests/csv/$', views.download_trainingrequests, name='download_trainingrequests'),
     url(r'^training_request/(?P<pk>\d+)/', include([
         url(r'^$', views.trainingrequest_details, name='trainingrequest_details'),
-        url(r'^edit/?$', views.TrainingRequestUpdate.as_view(), name='trainingrequest_edit'),
+        url(r'^edit/$', views.TrainingRequestUpdate.as_view(), name='trainingrequest_edit'),
     ])),
 
-    url(r'^trainees/?$', views.all_trainees, name='all_trainees'),
+    url(r'^trainees/$', views.all_trainees, name='all_trainees'),
 
-    url(r'^training_progresses/add/(?P<trainee_id>\d+)?$', views.TrainingProgressCreate.as_view(), name='trainingprogress_add'),
+    url(r'^training_progresses/add/(?P<trainee_id>\d+)?/$', views.TrainingProgressCreate.as_view(), name='trainingprogress_add'),
```

Comment:
Not sure how we should treat this URL. The group part `(?P<trainee_id>\d+)` seems to be optional, so with enforced trailing slash we theoretically could get `/workshops/training_progresses/add//` as a correct URL.